docs/proposals/parameter-distribution.md

```diff
@@ -0,0 +1,128 @@
+# Proposal for Parameter Distribution
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Design Details](#design-details)
+  - [Experiment API changes](#experiment-api-changes)
+  - [Correspondence for Katib Distributions and Framework Distributions](#correspondence-for-katib-distributions-and-framework-distributions)
+    - [Chocolate](#chocolate)
+    - [Goptuna](#goptuna)
+    - [Hyperopt](#hyperopt)
+    - [Optuna](#optuna)
+    - [Scikit-Optimize](#scikit-optimize)
+<!-- /toc -->
+
+## Summary
+This enhancement introduces `Distribution` to tuning parameters and remove redundantly `ParameterType`. 
+
+API field in the Experiment spec determine parameter type with distribution.  
+
+## Motivation
+Currently, Katib does not support determining a distribution for search space that samplers pick up parameters by users.
+
+Katib should be able to determine it by users since
+almost hyperparameter tuning algorithms (framework) can determine it by users.
+
+## Proposal
+We introduce a mechanism to determine a distribution for search space by users.
+That also means we introduce a mechanism to propagate distributions to suggestion-services and
+set them samplers of each framework.     
+
+## Design Details
+The proposal consists of a new Experiment API field and
+correspondence between Katib Distributions and Framework Distributions.
+
+### Experiment API changes
+We extend the Experiment API to introduce the new fields `Distribution` to configure the distribution for search space and
+remove the redundant fields `ParameterType`.
+
+```diff
+type ParameterSpec struct {
+	Name          string        `json:"name,omitempty"`
+- 	ParameterType ParameterType `json:"parameterType,omitempty"`
++ 	Distribution  Distribution  `json:"distribution,omitempty"`
+	FeasibleSpace FeasibleSpace `json:"feasibleSpace,omitempty"`
+}
+
+- type ParameterType string
++ type Distribution string
+
+const (
+- 	ParameterTypeUnknown     ParameterType = "unknown"
+- 	ParameterTypeDouble      ParameterType = "double"
+- 	ParameterTypeInt         ParameterType = "int"
+- 	ParameterTypeDiscrete    ParameterType = "discrete"
+- 	ParameterTypeCategorical ParameterType = "categorical"
++ 	UnknownDistribution         Distribution = "unknown"
++ 	CategoricalDistribution     Distribution = "categorical"
++ 	IntUniformDistribution      Distribution = "intUniform"
```

Comment:
Should we use camel case here? Personally prefer lower case `intuniform`

Reply:
SGTM